### PR TITLE
YUPANA-113 - remove duplicate entries from ip_addresses

### PR DESCRIPTION
**Pull request by kgaikwad**

### CODE REVIEW

**Comment by kgaikwad** [comment]
+ adding [ShimShtein](https://github.com/ShimShtein) into loop for review.

File: yupana/processor/report_slice_processor.py [hunk 1]
```diff
@@ -264,7 +264,12 @@ def _remove_empty_ip_addresses(
             host: dict, transformed_obj=copy.deepcopy(TRANSFORMED_DICT)):
         """Remove 'ip_addresses' field."""
         ip_addresses = host.get('ip_addresses')
-        if ip_addresses is None or ip_addresses:
+        if ip_addresses is None:
+            return [host, transformed_obj]
+        if ip_addresses:
```
Comment (patilsuraj767) [inline yupana/processor/report_slice_processor.py:269]:
```suggestion
        if ip_addresses is None or len(x) == len(set(x)): 
            return [host, transformed_obj]
        else:
```
Let us first check if list in unique, this way transformed_obj is only appended if ip_address list is really modified.

**Review by ShimShtein — CHANGES_REQUESTED** [review]
Nitpick: can we rename the method? I think `remove_empty_ip_addresses` no longer suits the actual work.
I suggest `transform_ip_addresses`

**Comment by kgaikwad** [comment]
> Nitpick: can we rename the method? I think `remove_empty_ip_addresses` no longer suits the actual work. I suggest `transform_ip_addresses`

@ShimShtein - Yeah, make sense. Fixed.

**Review by ShimShtein — APPROVED** [review]
LGTM

**Review by r14chandra — APPROVED** [review]
LGTM! 👍🏻

**Comment by kgaikwad** [comment]
Thanks @ShimShtein and @r14chandra for review!

**Pull request merged**